examples/apps/table-detail-tea/src/main.rs

```diff
@@ -0,0 +1,211 @@
+/// A Ratatui example that demonstrates the Elm architecture with a basic list - detail
+/// application.
+///
+/// This example runs with the Ratatui library code in the branch that you are currently
+/// reading. See the [`latest`] branch for the code which works with the most recent Ratatui
+/// release.
+///
+/// [`latest`]: https://github.com/ratatui/ratatui/tree/latest
+use std::io::stdout;
+use std::time::Duration;
+use std::{fmt, panic};
+
+use crossterm::event::{
+    self, DisableMouseCapture, EnableMouseCapture, Event, KeyCode, MouseEventKind,
+};
+use crossterm::ExecutableCommand;
+use fakeit::{address, contact, name};
+use ratatui::layout::{Constraint, Layout, Rect};
+use ratatui::style::{Modifier, Style};
+use ratatui::widgets::{Block, Padding, Paragraph, Row, Table, TableState};
+use ratatui::{DefaultTerminal, Frame};
+
+#[derive(Debug)]
+struct AppModel {
+    people: Vec<Person>,
+    table_state: TableState,
+}
+
+#[derive(Debug)]
+struct Person {
+    name: String,
+    address: Address,
+    email: String,
+}
+
+#[derive(Debug)]
+struct Address {
+    street: String,
+    city: String,
+    state: String,
+    zip: String,
+}
+
+#[derive(PartialEq, Copy, Clone)]
+enum Message {
+    SelectNext,
+    SelectPrevious,
+    Quit,
+}
+
+fn main() -> color_eyre::Result<()> {
+    let terminal = ratatui::init();
+    install_panic_hook();
+    stdout().execute(EnableMouseCapture)?;
+
+    let model = AppModel::default();
+    let result = model.run(terminal);
+
+    stdout().execute(DisableMouseCapture)?;
+    ratatui::restore();
+    result
+}
+
+pub fn install_panic_hook() {
```

Comment:
nit: add a comment here. a small note like:

```suggestion
/// Ensure that mouse mode is disabled if the app crashes
pub fn install_panic_hook() {
```